agent-part/src/test/kotlin/ExecRuntimeTest.kt

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright 2020 EPAM Systems
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.epam.drill.plugins.test2code
+
+import com.epam.drill.jacoco.*
+import kotlin.random.*
+import kotlin.test.*
+
+
+class ExecRuntimeTest {
+    private val threadLocal = ThreadLocal<Array<ExecDatum?>>()
+    private val className = "foo/bar/Foo"
+    private val testName = "test"
+
+    @AfterTest
+    fun removeThreadLocal() {
+        threadLocal.remove()
+    }
+
+    @Test
+    fun `getOrPut - add data with same test name`() {
+        val runtime = ExecRuntime { }
+        val first = runtime.getOrPut(testName) { arrayOfNulls(5) }
+        val second = runtime.getOrPut(testName) { arrayOfNulls(9) }
+        assertEquals(first, second)
+        assertEquals(5, first.size)
+        assertEquals(5, second.size)
+        assertTrue { runtime.collect().none() }
+    }
+
+    @Test
+    fun `putIndex - existing value`() {
```

Comment:
```suggestion
    fun `should put by Index when this index is existed`() {
```
putIndex - existing index, and collect ...
should be ... when/if ...